com.unity.netcode.gameobjects/Tests/Editor/Metrics/NetworkMetricsRegistrationTests.cs

```diff
@@ -0,0 +1,48 @@
+﻿#if MULTIPLAYER_TOOLS
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using NUnit.Framework;
+using Unity.Multiplayer.MetricTypes;
+using Unity.Multiplayer.NetStats.Dispatch;
+using Unity.Multiplayer.NetStats.Metrics;
+
+namespace Unity.Netcode.EditorTests.Metrics
+{
+    public class NetworkMetricsRegistrationTests
+    {
+        [Test]
```

Comment:
Just an idea - you could potentially use TestCaseSource here so that the Type is actually a parameter of the test instead of having a singular test for all metrics. It would make it so the Test Runner interface has a singular failing test that points to the specific metric, instead of having to dig into the logs of the failing test.

Reply:
I love this!